### PR TITLE
feat(FN-1728): auto reconciliation of zero action keying sheet rows

### PR DESCRIPTION
**Pull request by RLavender98**
## Introduction :pencil2:
If a keying sheet row has no actions we should automatically move it to status reconciled

## Resolution :heavy_check_mark:
- If a fee record has no payments, no fixed fee adjustment and no principal balance adjustment then it should move to status 'RECONCILED' when we generate keying data

### CODE REVIEW

File: dtfs-central-api/src/services/state-machines/fee-record/event-handlers/generate-keying-data/generate-keying-data.event-handler.ts
Comment (RLavender98) [inline dtfs-central-api/src/services/state-machines/fee-record/event-handlers/generate-keying-data/generate-keying-data.event-handler.ts:1]: There is a technical possibility that if the report only consists of fee records with zero fees paid to ukef for the period then the entire report could auto reconcile. I haven't added that check as I am waiting to hear back on if that is actually possible (it's not included in any ACs currently)

File: dtfs-central-api/src/services/state-machines/fee-record/event-handlers/generate-keying-data/generate-keying-data.event-handler.test.ts [hunk 1]
```diff
@@ -32,121 +33,224 @@ describe('handleFeeRecordGenerateKeyingDataEvent', () => {
   const aMatchingFeeRecord = () => FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withStatus('MATCH').build();
 
   beforeEach(() => {
-    jest.resetAllMocks();
     jest.mocked(calculateFixedFeeAdjustment).mockResolvedValue(10);
```
Comment (sw-mattw) [inline dtfs-central-api/src/services/state-machines/fee-record/event-handlers/generate-keying-data/generate-keying-data.event-handler.test.ts:36]: Is this line not redundant as the last mock implementation takes precedence?

**Review by sw-mattw — COMMENTED** [review]
Looks good, one query comment.

Are we waiting to hear back about that report state before merging or would that be covered by a different PR?

File: dtfs-central-api/src/services/state-machines/fee-record/event-handlers/generate-keying-data/generate-keying-data.event-handler.test.ts (hunk 1 above)
Comment (RLavender98) [reply dtfs-central-api/src/services/state-machines/fee-record/event-handlers/generate-keying-data/generate-keying-data.event-handler.test.ts:36]: Ah whoops yes, the line below should be calculatePrincipleBalanceAdjustment! Will update now

**Comment by RLavender98** [comment]
> Are we waiting to hear back about that report state before merging or would that be covered by a different PR?
Not heard back yet so I reckon a separate PR if it turns out it is needed

**Review by sw-mattw — APPROVED** [review]

**Pull request merged**